No longer supported ms-appx in ResourceDictionary source from v2.0

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://github.com/unoplatform/Uno/security/
-->

## Current behavior
I haven't upgraded recently and have been using V3.1.6. There was an error running after upgrading to v4.0 today, and this error starts with v2.0.

[App.xaml](https://github.com/Daoting/dt/blob/663845194bf64ac40df48eece864ecd18a1d8f78/Client/Common/App.xaml#L10)
The generated code for App_xxx.g.cs is the same as v3.1.6. The `ms-appx:///Dt.Base/Themes/Generic.xaml` is in class library `Dt.Base`. `GlobalStaticResources.g.cs` in `Dt.Base`:
```
		// Register ResourceDictionaries using ms-appx:/// syntax, this is called for external resources
		public static void RegisterResourceDictionariesBySource()
		{
			if(!_dictionariesRegistered)
			{
				_dictionariesRegistered = true;
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => Base_8668ae864c039084f93be925225e352e_ResourceDictionary);
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => Desk_458f85e1800a3bc55cf6266b03a4619f_ResourceDictionary);
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => Dialog_fe6438ceeae26ec062607d998e265911_ResourceDictionary);
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => FileList_dc082ac2952b60a36d8030f3a21fa2b3_ResourceDictionary);
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => FormView_c9c6509c9c92d49cf4fd18e2993f91d9_ResourceDictionary);
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => Generic_4a05253b87122aeaab4d7e007a8eed85_ResourceDictionary);
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => Global_c2decc6413d3e2cfb44cd22a13eca9ad_ResourceDictionary);
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => ListView_ad8b8c9037c489fdcec25d6b505a2f2d_ResourceDictionary);
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => Menu_9eb8f8b4d5f6bc938ccbcac51e90c201_ResourceDictionary);
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => Sketch_df88f108f6e46bc5500fc5dc06f32120_ResourceDictionary);
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => TreeView_fc2a8dfbffbe86d22dcb55c657b12818_ResourceDictionary);
				global::Uno.UI.ResourceResolver.RegisterResourceDictionaryBySource(uri: "ms-appx:///Dt.Base/", context: __ParseContext_, dictionary: () => Win_9be52796d2b6275cbc33bc9223e3f8d9_ResourceDictionary);
			}
		}
```
ResourceResolver.cs
```
		public static void RegisterResourceDictionaryBySource(string uri, XamlParseContext context, Func<ResourceDictionary> dictionary)
		{
			_registeredDictionariesByUri[uri] = dictionary;

			if (context != null)
			{
				// We store the dictionaries inside a ResourceDictionary to utilize the lazy-loading machinery
				var assemblyDict = _registeredDictionariesByAssembly.FindOrCreate(context.AssemblyName, () => new ResourceDictionary());
				var initializer = new ResourceDictionary.ResourceInitializer(dictionary);
				_assemblyRef++; // We don't actually use this key, we just need it to be unique
				assemblyDict[_assemblyRef] = initializer;
			}
		}
```
Is it because the path `ms-appx:///Dt.Base/` is incomplete? If the bug is easy to fix, please prioritize it because i can't start the project right now.

The output is:
```
Loaded assembly: System.Runtime.Serialization.dll [External]
12-23 06:47:34.177 D/Mono    ( 5701): DllImport searching in: '__Internal' ('(null)').
12-23 06:47:34.177 D/Mono    ( 5701): Searching for 'java_interop_jnienv_throw'.
12-23 06:47:34.177 D/Mono    ( 5701): Probing 'java_interop_jnienv_throw'.
12-23 06:47:34.177 D/Mono    ( 5701): Found as 'java_interop_jnienv_throw'.
12-23 06:47:35.941 E/mono    ( 5701): 
12-23 06:47:35.941 E/mono    ( 5701): Unhandled Exception:
12-23 06:47:35.941 E/mono    ( 5701): System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: Cannot locate resource from 'ms-appx:///Dt.Base/Themes/Generic.xaml'
12-23 06:47:35.941 E/mono    ( 5701):   at Uno.UI.ResourceResolver.RetrieveDictionaryForSource (System.String source, System.String currentAbsolutePath) [0x0003a] in C:\a\1\s\src\Uno.UI\UI\Xaml\ResourceResolver.cs:366 
12-23 06:47:35.941 E/mono    ( 5701):   at Dt.Shell.App.InitializeComponent () [0x0007f] in D:\Dt\Client\Dt.Shell.Droid\obj\Debug\100\g\XamlCodeGenerator\App_a9a445e08e07b50e74be0d8d3c09428f.g.cs:74 
12-23 06:47:35.941 E/mono    ( 5701):   at Dt.Shell.App..ctor () [0x00008] in D:\Dt\Client\Common\App.xaml.cs:29 
12-23 06:47:35.941 E/mono    ( 5701):   at App.Droid.Application+<>c.<.ctor>b__0_0 () [0x00000] in D:\Dt\Client\Dt.Shell.Droid\MainActivity.cs:24 
12-23 06:47:35.941 E/mono    ( 5701):   at Windows.UI.Xaml.NativeApplication..ctor (Windows.UI.Xaml.NativeApplication+AppBuilder appBuilder, System.IntPtr javaReference, Android.Runtime.JniHandleOwnership transfer) [0x00008] in C:\a\1\s\src\Uno.UI\UI\Xaml\NativeApplication.cs:36 
12-23 06:47:35.941 E/mono    ( 5701):   at App.Droid.Application..ctor (System.IntPtr javaReference, Android.Runtime.JniHandleOwnership transfer) [0x00000] in D:\Dt\Client\Dt.Shell.Droid\MainActivity.cs:24 
12-23 06:47:35.941 E/mono    ( 5701):   at (wrapper managed-to-native) System.Reflection.RuntimeConstructorInfo.InternalInvoke(System.Reflection.RuntimeConstructorInfo,object,object[],System.Exception&)
12-23 06:47:35.941 E/mono    ( 5701):   at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x00005] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:936 
12-23 06:47:35.941 E/mono    ( 5701):    --- End of inner exception stack trace ---
12-23 06:47:35.941 E/mono    ( 5701):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)
12-23 06:47:35.941 E/mono    ( 5701):   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)
12-23 06:47:35.942 E/mono-rt ( 5701): [ERROR] FATAL UNHANDLED EXCEPTION: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: Cannot locate resource from 'ms-appx:///Dt.Base/Themes/Generic.xaml'
12-23 06:47:35.942 E/mono-rt ( 5701):   at Uno.UI.ResourceResolver.RetrieveDictionaryForSource (System.String source, System.String currentAbsolutePath) [0x0003a] in C:\a\1\s\src\Uno.UI\UI\Xaml\ResourceResolver.cs:366 
12-23 06:47:35.942 E/mono-rt ( 5701):   at Dt.Shell.App.InitializeComponent () [0x0007f] in D:\Dt\Client\Dt.Shell.Droid\obj\Debug\100\g\XamlCodeGenerator\App_a9a445e08e07b50e74be0d8d3c09428f.g.cs:74 
12-23 06:47:35.942 E/mono-rt ( 5701):   at Dt.Shell.App..ctor () [0x00008] in D:\Dt\Client\Common\App.xaml.cs:29 
12-23 06:47:35.942 E/mono-rt ( 5701):   at App.Droid.Application+<>c.<.ctor>b__0_0 () [0x00000] in D:\Dt\Client\Dt.Shell.Droid\MainActivity.cs:24 
12-23 06:47:35.942 E/mono-rt ( 5701):   at Windows.UI.Xaml.NativeApplication..ctor (Windows.UI.Xaml.NativeApplication+AppBuilder appBuilder, System.IntPtr javaReference, Android.Runtime.JniHandleOwnership transfer) [0x00008] in C:\a\1\s\src\Uno.UI\UI\Xaml\NativeApplication.cs:36 
12-23 06:47:35.942 E/mono-rt ( 5701):   at App.Droid.Application..ctor (System.IntPtr javaReference, Android.Runtime.JniHandleOwnership transfer) [0x00000] in D:\Dt\Client\Dt.Shell.Droid\MainActivity.cs:24 
12-23 06:47:35.942 E/mono-rt ( 5701):   at (wrapper managed-to-native) System.Reflection.RuntimeConstructorInfo.InternalInvoke(System.Reflection.RuntimeConstructorInfo,object,object[],System.Exception&)
12-23 06:47:35.942 E/mono-rt ( 5701):   at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x00005] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:936 
12-23 06:47:35.942 E/mono-rt ( 5701):    --- End of inner exception stack trace ---
12-23 06:47:35.942 E/mono-rt ( 5701):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)
12-23 06:47:35.942 E/mono-rt ( 5701):   at (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)
```
<!-- Describe how the issue manifests. -->

## Expected behavior

<!-- Describe what the desired behavior would be. -->

## How to reproduce it (as minimally and precisely as possible)

<!-- Please provide a **MINIMAL REPRO PROJECT** and the **STEPS TO REPRODUCE**-->

## Workaround

<!-- Please provide steps to workaround this problem if possible -->

## Environment

<!-- For bug reports Check one or more of the following options with "x" -->

Nuget Package:
<!-- Please open issues on the project's repo if any, for instance:
       Uno.Material:       https://github.com/unoplatform/uno.material/issues
       Uno.Wasm.Bootstrap: https://github.com/unoplatform/uno.wasm.bootstrap/issues 
 -->
- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: <!-- Please specify -->

Nuget Package Version(s):

Affected platform(s):

- [x] iOS
- [x] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [x] Visual Studio 2019 (version:16.8.3)
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

## Anything else we need to know?

<!-- We would love to know of any friction, apart from knowledge, that prevented you from sending in a pull-request -->

I'm unable to build the app you link to, I get a lot of errors like the following:

```
1>        C:\src\repros\dt\Client\Dt.Core\Dt.Core.csproj : error NU1101: Unable to find package Dt.Cells. No packages exist with this id in source(s): [...]
1>    Done building target "ResolvePackageAssets" in project "Dt.Core.csproj" -- FAILED.
```

--- a/src/Common_ViewLibraryProps/Globbing.props
+++ b/src/Common_ViewLibraryProps/Globbing.props
@@ -1,0 +1,11 @@
+<Project ToolsVersion="Current">
+  <ItemGroup>
+    <Page Include="**\*.xaml" Exclude="bin\**\*.xaml;obj\**\*.xaml" />
+    <Compile Update="**\*.xaml.cs">
+      <DependentUpon>%(Filename)</DependentUpon>
+    </Compile>
+  </ItemGroup>
+  <ItemGroup>
+    <UpToDateCheckInput Include="**\*.xaml" Exclude="bin\**\*.xaml;obj\**\*.xaml" />
+  </ItemGroup>
+</Project>

--- a/src/SourceGenerators/Uno.UI.SourceGenerators/Content/Uno.UI.SourceGenerators.props
+++ b/src/SourceGenerators/Uno.UI.SourceGenerators/Content/Uno.UI.SourceGenerators.props
@@ -66,7 +66,8 @@
 
 		<CompilerVisibleItemMetadata Include="AdditionalFiles" MetadataName="SourceItemGroup" />
 		<CompilerVisibleItemMetadata Include="AdditionalFiles" MetadataName="Link" />
-		<CompilerVisibleItemMetadata Include="AdditionalFiles" MetadataName="FullPath" />
+        <CompilerVisibleItemMetadata Include="AdditionalFiles" MetadataName="FullPath" />
+        <CompilerVisibleItemMetadata Include="AdditionalFiles" MetadataName="Identity" />
 	  </ItemGroup>
 
 	</When>

--- a/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs
+++ b/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs
@@ -212,6 +212,10 @@ namespace Uno.UI.SourceGenerators.XamlGenerator
 					{
 						link = fullPath.Substring(definingProjectDirectory.Length);
 					}
+					else
+					{
+						link = projectItemInstance.GetMetadataValue("Identity");
+					}
 				}
 			}
 

--- a/src/Uno.UI-UnitTests-only.slnf
+++ b/src/Uno.UI-UnitTests-only.slnf
@@ -11,6 +11,7 @@
       "T4Generator\\T4Generator.csproj",
       "Uno.Foundation\\Uno.Foundation.csproj",
       "Uno.UI.FluentTheme\\Uno.UI.FluentTheme.csproj",
+      "Uno.UI.Tests.ViewLibraryProps\\Uno.UI.Tests.ViewLibraryProps.csproj",
       "Uno.UI.Tests.ViewLibrary\\Uno.UI.Tests.ViewLibrary.csproj",
       "Uno.UI.Tests\\Uno.UI.Tests.csproj",
       "Uno.UI\\Uno.UI.csproj",

--- a/src/Uno.UI.Tests.ViewLibraryProps/Themes/Button.xaml
+++ b/src/Uno.UI.Tests.ViewLibraryProps/Themes/Button.xaml
@@ -1,0 +1,8 @@
+﻿<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+					xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+					xmlns:win="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
+	<Style x:Key="GlobPropsMarginButtonStyle"  TargetType="Button">
+		<Setter Property="Margin"
+				Value="99,33,7,7" />
+	</Style>
+</ResourceDictionary>

--- a/src/Uno.UI.Tests.ViewLibraryProps/Uno.UI.Tests.ViewLibraryProps.csproj
+++ b/src/Uno.UI.Tests.ViewLibraryProps/Uno.UI.Tests.ViewLibraryProps.csproj
@@ -1,0 +1,68 @@
+<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
+	<PropertyGroup>
+		<TargetFrameworks>net461</TargetFrameworks>
+		<NoWarn>1998;4014;67</NoWarn>
+		<OutputPath>bin\$(Platform)\$(Configuration)</OutputPath>
+		<TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+		<ImportFrameworkWinFXTargets>false</ImportFrameworkWinFXTargets>
+	</PropertyGroup>
+
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+		<DebugSymbols>true</DebugSymbols>
+		<DebugType>full</DebugType>
+		<Optimize>false</Optimize>
+		<DefineConstants>TRACE;DEBUG;HAS_UMBRELLA_BINDING;IS_UNIT_TESTS;HAS_GEOCOORDINATE_WATCHER;HAS_GEOCOORDINATE;HAS_COMPILED_REGEX;HAS_ISTORAGEFILE;HAS_SEMAPHORE;HAS_FILE_IO;IS_UNO;HAS_CRIPPLEDREFLECTION;NET46</DefineConstants>
+		<TreatWarningsAsErrors>false</TreatWarningsAsErrors>
+	</PropertyGroup>
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
+		<DebugSymbols>true</DebugSymbols>
+		<DebugType>full</DebugType>
+		<Optimize>false</Optimize>
+		<DefineConstants>TRACE;HAS_UMBRELLA_BINDING;IS_UNIT_TESTS;HAS_GEOCOORDINATE_WATCHER;HAS_GEOCOORDINATE;HAS_COMPILED_REGEX;HAS_ISTORAGEFILE;HAS_SEMAPHORE;HAS_FILE_IO;IS_UNO;HAS_CRIPPLEDREFLECTION</DefineConstants>
+	</PropertyGroup>
+
+	<ItemGroup>
+		<PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
+		<PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="1.1.1" />
+		<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="1.1.1" />
+		<PackageReference Include="System.Collections.Immutable">
+			<Version>1.7.0</Version>
+		</PackageReference>
+		<PackageReference Include="Uno.SourceGenerationTasks" />
+		<PackageReference Include="Uno.Core" />
+	</ItemGroup>
+
+	<ItemGroup>
+		<Reference Include="PresentationCore" />
+		<Reference Include="PresentationFramework" />
+		<Reference Include="System" />
+		<Reference Include="System.Drawing" />
+		<Reference Include="System.Net.Http" />
+		<Reference Include="System.Numerics" />
+		<Reference Include="System.Numerics.Vectors" />
+		<Reference Include="System.Windows.Forms" />
+		<Reference Include="System.Xaml" />
+		<Reference Include="System.Xml" />
+		<Reference Include="System.Xml.Linq" />
+		<Reference Include="WindowsBase" />
+	</ItemGroup>
+
+	<PropertyGroup>
+		<UnoUIMSBuildTasksPath>$(MSBuildThisFileDirectory)..\SourceGenerators\Uno.UI.Tasks\bin\$(Configuration)_Shadow</UnoUIMSBuildTasksPath>
+		<UnoForceProcessPRIResource>true</UnoForceProcessPRIResource>
+	</PropertyGroup>
+
+	<Import Project="..\SourceGenerators\Uno.UI.SourceGenerators\Content\Uno.UI.SourceGenerators.props" />
+	<Import Project="..\SourceGenerators\Uno.UI.Tasks\Content\Uno.UI.Tasks.targets" Condition="'$(SkipUnoResourceGeneration)' == '' " />
+
+	<Target Name="BeforeBuildCleanup" BeforeTargets="XamlMarkupCompilePass1;_UnoSourceGenerator" />
+
+	<Import Project="..\Common.targets" />
+
+	<Import Project="..\Common_ViewLibraryProps\Globbing.props" />
+
+	<ItemGroup>
+		<ProjectReference Include="..\SourceGenerators\System.Xaml\Uno.Xaml.csproj" />
+		<ProjectReference Include="..\Uno.UI\Uno.UI.csproj" />
+	</ItemGroup>
+</Project>

--- a/src/Uno.UI.Tests/Uno.UI.Tests.csproj
+++ b/src/Uno.UI.Tests/Uno.UI.Tests.csproj
@@ -112,6 +112,7 @@
 		<ProjectReference Include="..\SourceGenerators\System.Xaml\Uno.Xaml.csproj" />
 		<ProjectReference Include="..\Uno.UI.FluentTheme\Uno.UI.FluentTheme.csproj" />
 		<ProjectReference Include="..\Uno.UI.Tests.ViewLibrary\Uno.UI.Tests.ViewLibrary.csproj" />
+		<ProjectReference Include="..\Uno.UI.Tests.ViewLibraryProps\Uno.UI.Tests.ViewLibraryProps.csproj" />
 		<ProjectReference Include="..\Uno.UI\Uno.UI.csproj" />
 	</ItemGroup>
 

--- a/src/Uno.UI.Tests/Windows_UI_Xaml/Controls/When_Source_And_Globbing_From_Included_File.xaml
+++ b/src/Uno.UI.Tests/Windows_UI_Xaml/Controls/When_Source_And_Globbing_From_Included_File.xaml
@@ -1,0 +1,18 @@
+﻿<UserControl
+    x:Class="Uno.UI.Tests.Windows_UI_Xaml.Controls.When_Source_And_Globbing_From_Included_File"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:local="using:Uno.UI.Tests.Windows_UI_Xaml.Controls"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    mc:Ignorable="d"
+    d:DesignHeight="300"
+    d:DesignWidth="400">
+	<UserControl.Resources>
+		<ResourceDictionary Source="ms-appx:///Uno.UI.Tests.ViewLibraryProps/Themes/Button.xaml"/>
+	</UserControl.Resources>
+
+    <Grid>
+
+    </Grid>
+</UserControl>

--- a/src/Uno.UI.Tests/Windows_UI_Xaml/Controls/When_Source_And_Globbing_From_Included_File.xaml.cs
+++ b/src/Uno.UI.Tests/Windows_UI_Xaml/Controls/When_Source_And_Globbing_From_Included_File.xaml.cs
@@ -1,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+// The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236
+
+namespace Uno.UI.Tests.Windows_UI_Xaml.Controls
+{
+	public sealed partial class When_Source_And_Globbing_From_Included_File : UserControl
+	{
+		public When_Source_And_Globbing_From_Included_File()
+		{
+			this.InitializeComponent();
+		}
+	}
+}

--- a/src/Uno.UI.Tests/Windows_UI_Xaml/Given_ResourceDictionary.cs
+++ b/src/Uno.UI.Tests/Windows_UI_Xaml/Given_ResourceDictionary.cs
@@ -720,5 +720,18 @@ namespace Uno.UI.Tests.Windows_UI_Xaml
 			var tb = page.ThemeDictionaryOnlyTextBlock;
 			Assert.AreEqual(Colors.MediumPurple, (tb.Foreground as SolidColorBrush).Color);
 		}
+
+		[TestMethod]
+		public void When_Source_And_Globbing_From_Included_File()
+		{
+			var ctrl = new When_Source_And_Globbing_From_Included_File();
+			var resources = ctrl.Resources;
+			Assert.IsTrue(resources.ContainsKey("GlobPropsMarginButtonStyle"));
+
+			var style = resources["GlobPropsMarginButtonStyle"] as Style;
+			var button = new Button();
+			button.Style = style;
+			Assert.AreEqual(new Thickness(99, 33, 7, 7), button.Margin);
+		}
 	}
 }

--- a/src/Uno.UI.sln
+++ b/src/Uno.UI.sln
@@ -207,6 +207,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Uno.UI.FluentTheme", "Uno.U
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Uno.UI.FluentTheme.Skia", "Uno.UI.FluentTheme\Uno.UI.FluentTheme.Skia.csproj", "{E7113F80-584B-4DF1-9BF0-166BB453B807}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Uno.UI.Tests.ViewLibraryProps", "Uno.UI.Tests.ViewLibraryProps\Uno.UI.Tests.ViewLibraryProps.csproj", "{68090F3E-ACE3-49D0-BC68-151F8A92A38B}"
+EndProject
 Global
 	GlobalSection(SharedMSBuildProjectFiles) = preSolution
 		SamplesApp\SamplesApp.UnitTests.Shared\SamplesApp.UnitTests.Shared.projitems*{04b1b5eb-d42e-47de-ada0-eb863e5574fd}*SharedItemsImports = 13
@@ -3598,6 +3600,48 @@ Global
 		{E7113F80-584B-4DF1-9BF0-166BB453B807}.Release|x64.Build.0 = Release|Any CPU
 		{E7113F80-584B-4DF1-9BF0-166BB453B807}.Release|x86.ActiveCfg = Release|Any CPU
 		{E7113F80-584B-4DF1-9BF0-166BB453B807}.Release|x86.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|ARM.ActiveCfg = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|ARM.Build.0 = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|ARM64.ActiveCfg = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|ARM64.Build.0 = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|iPhone.ActiveCfg = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|iPhone.Build.0 = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|iPhoneSimulator.ActiveCfg = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|iPhoneSimulator.Build.0 = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|x64.ActiveCfg = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|x64.Build.0 = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|x86.ActiveCfg = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Debug|x86.Build.0 = Debug|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|Any CPU.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|Any CPU.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|ARM.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|ARM.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|ARM64.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|ARM64.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|iPhone.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|iPhone.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|iPhoneSimulator.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|iPhoneSimulator.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|x64.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|x64.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|x86.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release_NoSamples|x86.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|Any CPU.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|ARM.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|ARM.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|ARM64.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|ARM64.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|iPhone.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|iPhone.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|iPhoneSimulator.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|iPhoneSimulator.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|x64.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|x64.Build.0 = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|x86.ActiveCfg = Release|Any CPU
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B}.Release|x86.Build.0 = Release|Any CPU
 	EndGlobalSection
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE
@@ -3691,6 +3735,7 @@ Global
 		{10807E33-4A73-4AA9-A8F2-B395CC555BD1} = {416684CF-A4E3-4079-B380-3FF0B00E433C}
 		{535339E2-6C01-4478-9889-F6E76B1C00F4} = {416684CF-A4E3-4079-B380-3FF0B00E433C}
 		{E7113F80-584B-4DF1-9BF0-166BB453B807} = {416684CF-A4E3-4079-B380-3FF0B00E433C}
+		{68090F3E-ACE3-49D0-BC68-151F8A92A38B} = {416684CF-A4E3-4079-B380-3FF0B00E433C}
 	EndGlobalSection
 	GlobalSection(ExtensibilityGlobals) = postSolution
 		SolutionGuid = {9B2608F4-D82B-4B72-B399-33E822DF01D0}

